src/main/java/seedu/user/ModuleRecord.java

```diff
@@ -29,22 +30,32 @@ public ModuleRecord() {
      *
      * @param module Module contains the graded module to be stored in the Transcript
      */
-    public void addModuleToRecord(GradedModule module) {
-        gradedModules.add(module);
-        moduleCodes.add(module.getModuleCode());
-        cap = calculateCapFromList();
-        TextUi.printAddedGrade(module.getModuleCode(), module.getGrade());
+    public void addModuleToRecord(GradedModule module) throws UniModsException {
+        boolean isModInRecords = checkModuleExists(module.getModuleCode());
+        if (isModInRecords) {
+            throw new UniModsException(TextUi.ERROR_MODULE_FOUND);
+        } else {
+            gradedModules.add(module);
+            moduleCodes.add(module.getModuleCode());
+            cap = calculateCapFromList();
+            TextUi.printAddedGrade(module.getModuleCode(), module.getGrade());
+        }
     }
 
     /**
      * Adds an ungraded Module to the Transcript.
      *
      * @param module Module contains the ungraded module to be stored in the Transcript
      */
-    public void addModuleToRecord(UngradedModule module) {
-        ungradedModules.add(module);
-        moduleCodes.add(module.getModuleCode());
-        TextUi.printAddedGrade(module.getModuleCode(), module.getGrade());
+    public void addModuleToRecord(UngradedModule module) throws UniModsException {
+        boolean isModInRecords = checkModuleExists(module.getModuleCode());
+        if (isModInRecords) {
+            throw new UniModsException(TextUi.ERROR_MODULE_FOUND);
+        } else {
+            ungradedModules.add(module);
+            moduleCodes.add(module.getModuleCode());
+            TextUi.printAddedGrade(module.getModuleCode(), module.getGrade());
+        }
```

Comment:
Perhaps refactoring this code with a parent class of Graded and Ungraded module might make it neater?